src/main/java/org/opensearch/flowframework/transport/ProvisionWorkflowTransportAction.java

```diff
@@ -145,7 +146,36 @@ protected void doExecute(Task task, WorkflowRequest request, ActionListener<Work
                             ActionListener.wrap(updateResponse -> {
                                 logger.info("updated workflow {} state to {}", request.getWorkflowId(), State.PROVISIONING);
                                 executeWorkflowAsync(workflowId, provisionProcessSequence, listener);
-                                listener.onResponse(new WorkflowResponse(workflowId));
+                                // update last provisioned field in template
+                                Template newTemplate = new Template.Builder().name(template.name())
+                                    .description(template.description())
+                                    .useCase(template.useCase())
+                                    .templateVersion(template.templateVersion())
+                                    .compatibilityVersion(template.compatibilityVersion())
+                                    .workflows(template.workflows())
+                                    .uiMetadata(template.getUiMetadata())
+                                    .user(template.getUser()) // Should we care about old user here?
```

Comment:
I'd agree here, the `user` in templates should be the one who created it. Curious though if it would be worth to track which user provisioned the template